### PR TITLE
Auto change color based on the theme

### LINKED ISSUE #24695: SVG images in the docs are not visible in dark mode
I'm sorry that I didn't pick up on this when I added the Furo theme. The SVG images in the docs do not support dark mode. For example, at https://docs.sympy.org/dev/modules/physics/vector/vectors.html:

Screen Shot 2023-02-09 at 2 57 42 PMScreen Shot 2023-02-09 at 2 57 48 PM

I'm not sure what the correct way to fix this is. 

Here's all the SVG files in the repo. Basically all of them are in the physics module:

```
$ git ls-files | grep svg
banner.svg
doc/ext/convert-svg-to-pdf.py
doc/src/logo/sympy-use-text.svg
doc/src/logo/sympy.svg
doc/src/modules/physics/mechanics/api/CylindricalJoint.svg
doc/src/modules/physics/mechanics/api/PinJoint.svg
doc/src/modules/physics/mechanics/api/PlanarJoint.svg
doc/src/modules/physics/mechanics/api/PrismaticJoint.svg
doc/src/modules/physics/mechanics/api/SphericalJoint.svg
doc/src/modules/physics/mechanics/api/WeldJoint.svg
doc/src/modules/physics/mechanics/api/joint_explanation.svg
doc/src/modules/physics/mechanics/examples/four_bar_linkage.svg
doc/src/modules/physics/mechanics/examples/pendulum_nonmin.svg
doc/src/modules/physics/mechanics/examples/rollingdisc.svg
doc/src/modules/physics/vector/kin_1.svg
doc/src/modules/physics/vector/kin_1pt.svg
doc/src/modules/physics/vector/kin_2.svg
doc/src/modules/physics/vector/kin_2pt.svg
doc/src/modules/physics/vector/kin_3.svg
doc/src/modules/physics/vector/kin_4.svg
doc/src/modules/physics/vector/kin_angvel1.svg
doc/src/modules/physics/vector/kin_angvel2.svg
doc/src/modules/physics/vector/kin_angvel3.svg
doc/src/modules/physics/vector/kin_rolling.svg
doc/src/modules/physics/vector/simp_rot.svg
doc/src/modules/physics/vector/vec_add.svg
doc/src/modules/physics/vector/vec_cross.svg
doc/src/modules/physics/vector/vec_dot.svg
doc/src/modules/physics/vector/vec_fix_notfix.svg
doc/src/modules/physics/vector/vec_mul.svg
doc/src/modules/physics/vector/vec_rep.svg
doc/src/modules/physics/vector/vec_simp_der.svg
doc/src/modules/vector/coordsys_rot.svg
```

### PR DESCRIPTION
**Pull request by tjstienstra**
#### References to other Issues or PRs

Proposal to fix #24695

#### Brief description of what is fixed or changed
This PR uses the proposal from [this comment](https://github.com/sympy/sympy/issues/24695#issuecomment-1723272283), where css can be used in the svg to autochange the color, by using raw html in the rst file.

#### Release Notes

NO ENTRY

### CODE REVIEW

**Comment by asmeurer** [comment]
Here's a preview https://output.circle-artifacts.com/output/job/0428db4a-62b2-407e-9f2e-b990bc978e6e/artifacts/0/doc/_build/html/modules/physics/vector/vectors.html

For me, there is no image that shows up, in either dark or light mode. 
image

File: doc/src/modules/physics/vector/vectors.rst [hunk 1]
```diff
@@ -13,10 +13,11 @@ Vector
...
 direction. Vectors in 3-space are often represented on paper as:
 
-.. image:: vec_rep.*
-   :height: 175
-   :width: 350
-   :align: center
+.. raw:: html
+
+    <svg class="align-center" viewbox="0 0 350 175" width="350px">
+        <use xlink:href="../../../_static/modules/physics/vector/vec_rep.svg#vec_rep_id"></use>
```
[the remark this answers is not in the document]
Comment (TJStienstra) [reply doc/src/modules/physics/vector/vectors.rst:19]: The `.. image` wraps the svg in `img` object, which seems to cause the css (`currentColor`) not to work anymore. This did work for me [here](https://tjstienstra.github.io/brim/guides/introduction.html#composing-a-model).

[the remark this answers is not in the document]
Comment (asmeurer) [reply doc/src/modules/physics/vector/vectors.rst:19]: I can confirm your example site works. Are you using any custom CSS on that site?

**Comment by TJStienstra** [comment]
> For me, there is no image that shows up, in either dark or light mode.

Hmmm, I would guess that it cannot find the image, though if I look in the CircleCI build the file is in the expected location. Bit strange that it does not work here, as it did work for me when building the documentation locally.

**Comment by TJStienstra** [comment]
If we would be able by the way to get this working we could opt for creating a [custom directive](https://www.sphinx-doc.org/en/master/development/tutorials/helloworld.html) specifically for SVGs.

**Comment by TJStienstra** [comment]
Came across the solution I just implemented [here](https://github.com/sphinx-doc/sphinx/issues/2240#issuecomment-492200590).

File: doc/src/modules/physics/vector/vectors.rst (hunk 1 above)
[the remark this answers is not in the document]
Comment (TJStienstra) [reply doc/src/modules/physics/vector/vectors.rst]: No, I'm just changing the `#000000` color references in the image to `currentColor`, which is a CSS variable.

[the remark this answers is not in the document]
Comment (TJStienstra) [reply doc/src/modules/physics/vector/vectors.rst]: The other solution with `:file:` does [actually work](https://output.circle-artifacts.com/output/job/77eab45b-b8fd-4228-bf13-ab66ca3ac746/artifacts/0/doc/_build/html/modules/physics/vector/vectors.html).

**Comment by asmeurer** [comment]
Nice, I can confirm this does work, with all 6 combinations of system/page light/dark/auto settings. https://output.circle-artifacts.com/output/job/77eab45b-b8fd-4228-bf13-ab66ca3ac746/artifacts/0/doc/_build/html/modules/physics/vector/vectors.html

So now we need to change all the remaining SVGs in the docs, and also document this somewhere. 

We might also consider making some upstream improvements or at least open issues in Furo and/or Sphinx about this.

**Comment by TJStienstra** [comment]
Just took the time to go through all SVGs. During this process, I also cleaned up some of the SVGs in terms of code. The primary change in the commits so far is still making the images dark theme compatible by just letting colors switch from black to white. The 'kin_angvel2.svg' image has also undergone a subtle color adjustment.

It's worth noting that the SVGs referenced in the docstrings, such as the images from the joints framework, do not currently auto-adjust their colors. In my opinion, this isn't a significant concern as they remain generally readable. Nevertheless, if there is auto-changing the color is desired, it is possible. However, the disadvantage is that the entire file paths within docstrings  need to be specified, which is cumbersome. Let me know if this is a required change.

**Comment by asmeurer** [comment]
What do these docstring images look like in dark mode?

**Comment by TJStienstra** [comment]
They are all on [this page](https://docs.sympy.org/dev/modules/physics/mechanics/api/joint.html).

File: sympy/physics/mechanics/joint.py [hunk 2]
```diff
@@ -556,7 +556,8 @@ def create_symbol(number):
 class PinJoint(Joint):
     """Pin (Revolute) Joint.
 
-    .. image:: PinJoint.svg
+    .. raw:: html
+        :file: ../../../doc/src/modules/physics/mechanics/api/PinJoint.svg
```
Comment (TJStienstra) [inline sympy/physics/mechanics/joint.py:560]: Changed it for the `PinJoint`. Don't really like specifying the path here, but it does fix it.

**Comment by asmeurer** [comment]
Is there an upstream issue in Sphinx to make it so that we don't have to use these workarounds?

**Comment by TJStienstra** [comment]
> Is there an upstream issue in Sphinx to make it so that we don't have to use these workarounds?

I would say that it is related to [this issue](https://github.com/sphinx-doc/sphinx/issues/2240).

**Comment by asmeurer** [comment]
Looks like the diagonal lines at the top of the svg on this page aren't modified correctly https://output.circle-artifacts.com/output/job/7e1d9056-97bf-4724-9663-09d92b9894b5/artifacts/0/doc/_build/html/modules/physics/mechanics/examples/lin_pend_nonmin_example.html

File: doc/src/modules/physics/vector/kin_angvel2.svg
Comment (asmeurer) [inline doc/src/modules/physics/vector/kin_angvel2.svg:1]: I'm concerned about the color contrast of this SVG, especially in dark mode. Please make sure that the text color and background colors are WGAC AA compliant. You can use https://accessible-colors.com/ to check.

Comment (tjstienstra) [reply doc/src/modules/physics/vector/kin_angvel2.svg:1]: Awesome tool. Thank you for the suggestion. The contrast ratio is now 4.54 in dark theme and 4.62 in light theme. Addressed in 0b4fa718242f1b117396cd6662e11be5e195749b

**Comment by asmeurer** [comment]
This looks good to me now, thanks.

**Pull request merged**